float32[] id
float32[] xpos
float32[] ypos
